lib/Models/CsvCatalogItem.js

```diff
@@ -210,6 +226,9 @@ CsvCatalogItem.prototype.startPolling = function() {
             }
           });
       }
+      // update isPolling - if the item is disabled then we are not polling
+      polling.isPolling = false;
+
       // Note this means the timer keeps going even when you remove (disable) the item,
       // but it doesn't actually request new data any more.
       // If the item is re-enabled, the same timer just starts picking it up again.
```

Comment:
Is it that important that the timer picks up from the last one when item is disabled and enabled again? I don't see much point of that here since data is not being updated in the background anyway. Instead, when an item is disabled, we can clear the timeout and only keep polling as long as the item is enabled? Especially if we have a lot of datasets with polling and each have a timer constantly going in the background even after they have been disabled

Reply:
The actual model layer code in this file for performing the polling was here already, I just added a few variables. It's definitely less efficient than it could be, but I decided to focus on the UI stuff rather than rewriting this bit since it's most likely going to be rewritten in less than 6 months anyway.

Reply:
We should absolutely do what you're suggesting, I'm just not sure if we should do it now.

Reply:
It might be something that should go into a different issue, but for the sake of trying, here's just an idea, what about subscribe to `isEnabled` in `CSVCatalogItem` and startPolling every time `isEnabled` is true? I do not know what it was done the original way, maybe you know the big picture better than me. 

If not we should create a seperate issue to document this & address it in near future.

Reply:
@kring any idea why it might have been originally written this way? because I don't know why either haha